File: 减小数组使其满足条件的最小K值.py
class Solution:
    def minimumK(self, nums: List[int]) -> int:
        left, right = ceil(sqrt(len(nums))) - 1, max(nums) + 1
        while left + 1 < right:
            mid = (left + right) >> 1
            cnt = 0
            for num in nums:
                cnt += (num + mid - 1) // mid
            if cnt <= mid * mid:
                right = mid
            else:
                left = mid
        return left + 1

File: 减小数组使其满足条件的最小K值.cpp
class Solution {
public:
    int minimumK(vector<int>& nums) {
        int left = ceil(sqrt(nums.size())) - 1, right = *max_element(nums.begin(), nums.end()) + 1;
        while (left + 1 < right) {
            int mid = (left + right) >> 1, cnt = 0;
            for (int num : nums) {
                cnt += (num + mid - 1) / mid;
            }
            if (cnt <= (long long)mid * mid) {
                right = mid;
            } else {
                left = mid;
            }
        }
        return left + 1;
    }
};
